kotest-extensions/src/jvmMain/kotlin/io/kotest/extensions/system/wireListeners.kt

```diff
@@ -54,6 +55,38 @@ class SystemOutWireListener(private val tee: Boolean = true) : TestListener {
    fun output(): String = String(buffer.toByteArray())
 
    override suspend fun beforeTest(testCase: TestCase) {
+      redirectStandardOut()
+   }
+
+   override suspend fun afterTest(testCase: TestCase, result: TestResult) {
+      restoreStandardOutput()
+   }
+
+   override suspend fun beforeContainer(testCase: TestCase) {
+      if (testCase.type == TestType.Container) redirectStandardOut()
+   }
+
+   override suspend fun afterContainer(testCase: TestCase, result: TestResult) {
+      if (testCase.type == TestType.Container) restoreStandardOutput()
+   }
+
+   override suspend fun beforeEach(testCase: TestCase) {
+      if (testCase.type == TestType.Test) redirectStandardOut()
+   }
+
+   override suspend fun afterEach(testCase: TestCase, result: TestResult) {
+      if (testCase.type == TestType.Test) restoreStandardOutput()
+   }
+
+   override suspend fun beforeAny(testCase: TestCase) {
+      redirectStandardOut()
+   }
+
+   override suspend fun afterAny(testCase: TestCase, result: TestResult) {
+      restoreStandardOutput()
```

Comment:
We probably only need to call redirectStandardOutrestoreStandardOutputresetnow on one of the beforeAny / beforeTest cases as that will cover it for every test.